torch/csrc/utils/python_numbers.h

```diff
@@ -107,7 +107,7 @@ inline bool THPUtils_unpackBool(PyObject* obj) {
 
 inline bool THPUtils_checkDouble(PyObject* obj) {
 #ifdef USE_NUMPY
-  if (torch::utils::is_numpy_scalar(obj) && ! PyBool_Check(obj)) {
+  if (torch::utils::is_numpy_scalar(obj)) {
```

Comment:
removing the bool check here caused test failures:

```
FAIL: test_default_collate_dtype (__main__.TestDataLoader)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/bvaughan/repos/pytorch/test/test_dataloader.py", line 1531, in test_default_collate_dtype
    self.assertEqual(collated.dtype, torch.bool)
  File "/home/bvaughan/repos/pytorch/test/common_utils.py", line 841, in assertEqual
    super(TestCase, self).assertEqual(x, y, message)
AssertionError: torch.int64 != torch.bool :
```

Reply:
On master we don't have the bool check: https://github.com/pytorch/pytorch/blob/244b0bd1a5a64da370f44fca7ba4a73e296dd4b0/torch/csrc/utils/python_numbers.h#L102-L114 , so why is the bool check necessary now?

Reply:
sorry I think I commented on the wrong line. it looks like the tests were broken where we removed the PyBool_Check in is_numpy_int, in this same commit, (tensor_numpy.cpp).

Master is wrong though, so we can't rely on what master did to tell us what's correct.

Reply:
> Master is wrong though, so we can’t rely on what master did to tell us what’s correct 

Master isn’t completely wrong because the code works for applications other than the use case this PR is trying to fix. I’m not suggesting that master is correct on this; I’m probing about why we are doing things to better understand what is going on